app/Models/Leaderboard.php

```diff
@@ -225,6 +233,34 @@ public function scopeVisible(Builder $query): Builder
         return $query->where('DisplayOrder', '>=', 0);
     }
 
+    /**
+     * @param Builder<Leaderboard> $query
+     * @return Builder<Leaderboard>
+     */
+    public function scopeWithTopEntry(Builder $query)
+    {
+    return $query->addSelect(['top_entry_id' => function ($subQuery) {
+        $subQuery->select('le.id')
+            ->from('leaderboard_entries as le')
+            ->join('UserAccounts as u', 'u.id', '=', 'le.user_id')
+            ->whereColumn('le.leaderboard_id', 'LeaderboardDef.ID')
+            ->whereNull('u.unranked_at')
+            ->where('u.Untracked', 0)
+            ->whereNull('u.banned_at')
+            ->whereNull('le.deleted_at')
+            ->orderByRaw('
+                CASE
+                    WHEN (SELECT LowerIsBetter FROM LeaderboardDef WHERE ID = le.leaderboard_id) = 1 THEN le.score
+                    ELSE -le.score
+                END ASC
+            ')
+            ->orderBy('le.updated_at', 'ASC')
+            ->limit(1);
+    }])->with(['topEntry' => function ($query) {
+        $query->with('user');
+    }]);
+}
```

Comment:
nit: There's some whitespace weirdness happening here that we should probably fix up.

Reply:
See if that's better?